networking/portfwd.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networking
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/coreos/rocket/Godeps/_workspace/src/github.com/coreos/go-iptables/iptables"
+)
+
+func (e *containerEnv) forwardPorts(fps []ForwardedPort, defIP net.IP) error {
+	if len(fps) == 0 {
+		return nil
+	}
+
+	ipt, err := iptables.New()
+	if err != nil {
+		return err
+	}
+
+	// Create a separate chain for this container. This helps with debugging
+	// and makes it easier to cleanup
+	chain := e.portFwdChain()
+
+	if err = ipt.NewChain("nat", chain); err != nil {
+		return err
+	}
+
+	rule := e.portFwdRuleSpec(chain)
+
+	// outside traffic hitting this hot
+	if err = ipt.AppendUnique("nat", "PREROUTING", rule...); err != nil {
```

Comment:
I guess we can but it should be very informative. When dumping iptables, I usually have trouble mentally parsing because of all the output. So the comments should really be pointing out something insightful or they'll detract more than they'll help.